### ISSUE TITLE
Level-dependent parameterization

### ISSUE THREAD

**Issue by effigies** [poldracklab/fitlins#174]
One thing that FitLins should be useful for (especially after #171) is to swap in and out different tools at different levels, allowing users to specify a precise methodological difference when systematically comparing tools.

This is going to require some level-dependent parameterization, such as what estimator to use at each level, and there may be other issues, such as smoothing or resampling (e.g., from volume to surface) that also make sense to separate out.

One thought is to write a parser (not sure if this is manageable with `argparse`) that allows for nested option namespaces:

```
fitlins <global flags>
    --level=run <run-level flags> \
    --level=dataset <dataset-level flags> \
    --level=all <more global flags>
```

Or a JSON specification that will do the same.

Any other thoughts?

**Comment by adelavega** [comment]
That seems reasonable, although maybe not super high priority

**Comment by effigies** [comment]
Agreed. "High effort" is my code for "low priority".
